themes/default/content/blog/advanced-aws-networking-part-1/index.md

```diff
@@ -0,0 +1,276 @@
+        appliance_mode_support="enable",
+        tags={
+            "Name": f"{name}",
+        },
+    ),
+    # We can only have one attachment per VPC, so we need to tell Pulumi
+    # explicitly to delete the old one before creating a new one:
+    pulumi.ResourceOptions(
+        delete_before_replace=True,
+        depends_on=[self.vpc],
+        parent=self,
+    )
+)
+```
+
+Then add a route to the Transit Gateway route table for the _spoke_ VPCs that routes all traffic outside of the VPC to the inspection VPC (or to be precise, the Transit Gateway attachment in the inspection VPC). This route only needs to be added once as it will apply to all spoke VPCs, so we add it here in the inspection VPC since that's where we create the Transit Gateway attachment:
+
+```python
+aws.ec2transitgateway.Route(
+    f"{name}-default-spoke-to-inspection",
+    aws.ec2transitgateway.RouteArgs(
+        destination_cidr_block="0.0.0.0/0",
+        transit_gateway_attachment_id=self.tgw_attachment.id,
+        transit_gateway_route_table_id=args.spoke_tgw_route_table_id,
+    ),
+    opts=pulumi.ResourceOptions(
+        parent=self,
+    )
+)
+```
+
+Next, add routes within the inspection VPC's isolated subnets that route traffic to the internet through the NAT gateway. The Transit Gateway:
+
+```python
+for subnet_id in isolated_subnet_ids:
+    route_table = aws.ec2.get_route_table(
+        subnet_id=subnet_id
+    )
+
+    aws.ec2.Route(
+        f"{self.name}-route-{subnet_id}-to-tgw",
+        aws.ec2.RouteArgs(
+            route_table_id=route_table.id,
+            destination_cidr_block="0.0.0.0/0",
+            nat_gateway_id=self.nat_gateway.id,
+        ),
+        pulumi.ResourceOptions(
+            parent=self,
+        ),
+    )
+```
+
+Finally, you'll need to add routes within the inspection VPC's public subnets (where the NAT gateways live) that route return traffic from the internet back through the Transit Gateway. This is how the inspection VPC will be able to route traffic to the spoke VPCs:
+
+```python
+for subnet_id in public_subnet_ids:
+    route_table = aws.ec2.get_route_table(
+        subnet_id=subnet_id
+    )
+
+    aws.ec2.Route(
+        f"{self.name}-route-{subnet_id}-to-tgw",
+        aws.ec2.RouteArgs(
+            route_table_id=route_table.id,
+            destination_cidr_block=self.args.supernet_cidr_block,
+            transit_gateway_id=self.args.tgw_id,
+        ),
+        pulumi.ResourceOptions(
+            depends_on=[self.tgw_attachment],
+            parent=self,
+        ),
+    )
+```
+
+You need to both sets of these routes within an [apply](https://www.pulumi.com/docs/intro/concepts/inputs-outputs/#apply) because of we need to access the raw values of each of the subnet ids. Apply can be conceptually difficult for beginners in Pulumi to grasp, but it essentially means "once you know all of the values that I need, pass them on to this function".
```

Comment:
```suggestion
You need both sets of these routes in an [apply](https://www.pulumi.com/docs/intro/concepts/inputs-outputs/#apply) because we need to access the raw values of each of the subnet ids. Apply means once you know all of the values needed, pass them on to this function.
```

Reply:
Fixed.